purchase_invoice.py

```diff
@@ -875,7 +884,188 @@ def parse_invoice(self,infile,account=None,paid_by_submitter=False,
+                    self.supplier = values['-supplier-name-']
+            if '-no-' in values:
+                self.no = values['-no-']
+            if '-date-' in values:
+                date = utils.convert_date4(values['-date-'])
+                if date:
+                    self.date = date
+            if '-vat-' in values:
+                self.vat[self.default_vat] = utils.read_float(values['-vat-'])
+            if '-gross-' in values:
+                self.totals[self.default_vat] = utils.read_float(values['-gross-']) - self.vat[self.default_vat]
+            if '-skonto-' in values and values['-skonto-']:
+                self.skonto = utils.read_float(values['-skonto-'])
+            if '-paid-' in values and values['-paid-']:
+                self.paid_by_submitter = True
+            if '-remarks-' in values:
+                self.remarks = values['-remarks-']
+        else:
+            return None
+        self.compute_total()
+        account = None
+        accounts = self.company.leaf_accounts_for_credit
+        account_names = [acc['name'] for acc in accounts]
+        if default_account:
+            for acc in account_names:
+                if default_account in acc:
+                    account = acc
+        if not account:
+            pinvs = self.company.purchase_invoices[self.supplier]
+            paccs = [pi['expense_account'] for pi in pinvs if 'expense_account' in pi]
+            paccs = list(set(paccs))
+            for acc in paccs:
+                try:
+                    account_names.remove(acc)
+                except Exception:
+                    pass
+            account_names = paccs + account_names
+            title = 'Buchungskonto wählen'
+            msg = 'Bitte ein Buchungskonto wählen\n'
+            account = easygui.choicebox(msg, title, account_names)
+            if not account:
+                return None
+        self.assign_default_e_items({self.default_vat: account})
+        return self
+
+    def parse_invoice_json(self, invoice_json, account=None, paid_by_submitter=False, given_supplier=None, is_test=False):
+        self.extract_items = self.update_stock
+        supps = dict(PurchaseInvoice.suppliers)
+        if self.company.name != 'Laden':
+            del supps['Rechnung']
+        try:
+            self.supplier = [el.get('value') for el in invoice_json['entities'] if el.get('type') == 'supplier_name'][0]
+            for supplier, info in supps.items():
+                if supplier in self.supplier or self.supplier in supplier:
+                    self.parser = supplier
+                    if info['raw']:
+                        self.raw = True
+                    print("Verwende Rechnungsparser für ", self.supplier)
+                    if given_supplier and self.supplier != given_supplier:
+                        print("abweichend von PreRechnung: ", given_supplier)
+                    self.multi = info['multi']
```

Comment:
you do not need all this stuff. This was for selecting a supplier-specific hand-written parser.

Reply:
I have removed that